### ISSUE TITLE
Smart Tab support

### ISSUE THREAD

**Issue by plinss** [zed-industries/zed#4307]
### Check for existing issues

- [X] Completed

### Describe the feature

Please add support for "Smart Tabs", i.e. using tabs for indentation and spaces for alignment. This also implies adding support for auto-indenting to add alignment spaces.

For reference, here's an equivalent request for VSCode: https://github.com/Microsoft/vscode/issues/33974 

SublimeText supports this behavior well and this is the main blocker for me switching from SublimeText to Zed.

Given the following Python code (using `>` for a tab and `.` for a space:
```
>   def func(one, two):
```
pressing enter after `one,` should result in the following:
```
>   def func(one,
>   .........two):
```

Currently Zed results in:
```
>   def func(one,
>   >   two):
```

even worse, when tying that function and pressing enter after `one,`, Zed indents as:
```
>   def func(one,
>   
```
when then manually adding spaces after the tab, Zed then removes the tab and most of the spaces when you close the parenthesis after `two`. e.g.:
```
>   def func(one,
..two)
```

After correcting that, Zed mis-indents the next line by adding a tab for every space in the preceding line, e.g.:
```
>   def func(one, 
>   .........two):
>   >   >   >   >   >   >   >   >   >   >   											
```
instead of:
```
>   def func(one, 
>   .........two):
>   >   
```

**Comment by plinss** [comment]
Feature is still not implemented. Feature is still desired.

Stale bots are generally not helpful and can be quite hostile to your community.

**Comment by zelenenka** [comment]
@plinss you're right, the feature is indeed not implemented. the bot ignored the configuration that told it not to touch feature requests. it's a bug: https://github.com/actions/stale/issues/1302. sorry about that.
